code/src/reducers/ui.js

```diff
@@ -0,0 +1,15 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const ui = createSlice({
+  name: 'ui',
+  initialState: {
+    Loading: false
+  },
+  reducers: {
+    setLoading: (store, action) => {
+      store.Loading = action.payload
+    }
+  }
+})
+
+export default ui
```

Comment:
We also have this as a separate reducer, but since it's only one action, maybe it could live inside the labyrinth reducer.